### PR TITLE
add Contact Name to blocked number.

### PR DESCRIPTION
**Pull request by wilsonrc**
### Description

Add Contact Name to Manage Blocked Numbers Screen.

I created a new method to load the blocked numbers with the contact name to avoid breaking the other uses of the getBlockedNumber function.

### CODE REVIEW

File: commons/src/main/kotlin/com/simplemobiletools/commons/extensions/Context.kt [hunk 1]
```diff
@@ -992,6 +992,47 @@ fun Context.isDefaultDialer(): Boolean {
...
+            BlockedNumbers.COLUMN_E164_NUMBER,
+        )
+
+        queryCursor(uri, projection) { cursor ->
+            val id = cursor.getLongValue(BlockedNumbers.COLUMN_ID)
+            val number = cursor.getStringValue(BlockedNumbers.COLUMN_ORIGINAL_NUMBER) ?: ""
+            val normalizedNumber = cursor.getStringValue(BlockedNumbers.COLUMN_E164_NUMBER) ?: number
+            val comparableNumber = normalizedNumber.trimToComparableNumber()
+
+            val contactName = contacts[number] ?: getString(R.string.unknown)
```
Comment (tibbi) [inline commons/src/main/kotlin/com/simplemobiletools/commons/extensions/Context.kt:1028]: does this also handle settings like Start name with surname properly?

Comment (wilsonrc) [reply commons/src/main/kotlin/com/simplemobiletools/commons/extensions/Context.kt:1028]: I think So. This is using the name property  `Contact.name` that uses the `getNameToDisplay()` which internally handles the start name with surname preference.

Comment (wilsonrc) [reply commons/src/main/kotlin/com/simplemobiletools/commons/extensions/Context.kt:1028]: I think So. This is using the name property  `Contact.name` that uses the `getNameToDisplay()`, which internally handles the start name with surname preference.

**Comment by tibbi** [comment]
test it please, dont just think. Anyway, updating commons is a bit more difficult now, you have to update target SDK to 34 of the given app too, ideally rewrite to viewbinding aswell.

File: commons/src/main/kotlin/com/simplemobiletools/commons/extensions/Context.kt (hunk 1 above)
Comment (wilsonrc) [reply commons/src/main/kotlin/com/simplemobiletools/commons/extensions/Context.kt:1032]: @tibbi I tested it, and it handles start with surname correctly.

**Pull request merged**